subprojects/core/src/main/java/org/gradle/api/internal/DefaultDomainObjectCollection.java

```diff
@@ -140,8 +140,8 @@ public void all(Action<? super T> action) {
     /**
      * Returns true if, and only if, the store is empty AND we know that we
      * can query its size in constant time. Otherwise, it returns false, which
-     * doesn't mean that the store is empty, but just that we cannot determine
-     * in constant time.
+     * doesn't mean that the store is empty or not empty, but just that we cannot determine
```

Comment:
I'd get rid of all the double negatives:

Otherwise it returns false, which means that the collection may contain elements or may be empty.